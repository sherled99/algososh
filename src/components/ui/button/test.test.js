import renderer from "react-test-renderer"; // Импортируйте react-test-renderer
import { Button } from "./button";

describe("Button Component", () => {
  it("renders a button with text", () => {
    const tree = renderer.create(<Button text="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a button without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a disabled button when loader is active", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a loading indicator when loader is active", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls the callback function when the button is clicked", () => {
    const onClickMock = jest.fn();
    const tree = renderer.create(<Button text="Click me" onClick={onClickMock} />);
    // Симулируйте клик на кнопку здесь
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
