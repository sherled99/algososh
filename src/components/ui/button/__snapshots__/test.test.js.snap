// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component calls the callback function when the button is clicked 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button Component renders a button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button Component renders a button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Component renders a disabled button when loader is active 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Component renders a loading indicator when loader is active 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
